### DISCUSSION TITLE
Using clang-format for consistent code formatting

### DISCUSSION THREAD

**Discussion by stv0g** [sogno-platform/dpsim#274] [Ideas]
Hi @dinkelbachjan @m-mirz @martinmoraga @leonardocarreras,

We've recently started to use [clang-format](https://clang.llvm.org/docs/ClangFormat.html) to automatically format the VILLASnode C++ code.

I would love to introduce clang-format also to DPsim. It seems we have currently various different styles.
Clang-format could harmonize all of that by running a single command.

What are your opinions on this? If the the response is positive, I could prepare an MR implementing this change.

**Comment by m-mirz** [comment]
I think this is a great idea and would support it :+1:

**Comment by stv0g** [comment]
Any other opinions? @dinkelbachjan @martinmoraga @leonardocarreras?

I think that there are a few larger PRs currently open? We should at least wait until those are merged?

**Comment by martinmoraga** [comment]
@stv0g 

I support your idea too! 
But I wouldn't wait until all the other PRs have been merged as it can a while :D

**Comment by stv0g** [comment]
Hi @dinkelbachjan,

It would nice to hear your opinion on this as well before I start preparing a PR :)

**Comment by leonardocarreras** [comment]
Hi @stv0g ! I think this is a nice improvement... 

Do you think of it like as a pre-commit hook and progressively improve or apply it to the existing codebase in one go?

**Reply by stv0g** [reply]
I would do it onnce for all manually and than redo it manually once im a while. I personally also use a pre-commit hook. But I dont want tobemforce this for newcomers to keep the entry burden low.

**Comment by stv0g** [comment]
Lets continue the discussion in https://github.com/sogno-platform/dpsim/pull/278
